api_app/script_analyzers/file_analyzers/strings_info.py

```diff
@@ -24,58 +27,32 @@ def set_config(self, additional_config_params):
-            if self.rank_strings:
-                rank_command = ["rank_strings", "-l", str(self.max_no_of_strings)]
-                p2 = Popen(rank_command, stdin=p1.stdout, stdout=PIPE, stderr=PIPE)
-                out, err = p2.communicate()
-                output_rankstrings = out.decode()
-
-                if p2.returncode != 0:
-                    raise AnalyzerRunException(
-                        f"rank_strings return code is {p2.returncode}. Error: {err}"
-                    )
-                if len(output_rankstrings) == self.max_no_of_strings:
-                    results["exceeded_max_number_of_strings"] = True
-                results["ranked_strings"] = [
-                    s[: self.max_chars_for_string]
-                    for s in output_rankstrings.split("\n")
-                ]
-
-            else:
-                out, err = p1.communicate()
-                output_flarestrings = out.decode()
-
-                if p1.returncode != 0:
-                    raise AnalyzerRunException(
-                        f"flarestrings return code is {p1.returncode}. Error: {err}"
-                    )
-                if len(output_flarestrings) >= self.max_no_of_strings:
-                    results["exceeded_max_number_of_strings"] = True
-                results["flare_strings"] = [
-                    s[: self.max_chars_for_string]
-                    for s in output_flarestrings.split("\n")[: self.max_no_of_strings]
-                ]
-
-        except SoftTimeLimitExceeded as e:
-            error_message = (
-                f"job_id:{self.job_id} analyzer:{self.analyzer_name} md5:{self.md5}"
-                f"filename:{self.filename}. Soft Time Limit Exceeded Error: {e}"
-            )
-            logger.error(error_message)
-            self.report["errors"].append(str(e))
-            self.report["success"] = False
-            # we should stop the subprocesses...
-            # .. in case we reach the time limit for the celery task
-            if p1:
-                p1.kill()
-            if p2:
-                p2.kill()
-
-        return results
+        # get binary
+        binary = get_binary(self.job_id)
+        # make request data
+        fname = str(self.filename).replace("/", "_").replace(" ", "_")
+        args = ["flarestrings", f"@{fname}"]
+        req_data = {
+            "args": args,
+            "timeout": self.timeout,
+        }
+        req_files = {fname: binary}
+        result = self._docker_run(req_data, req_files)
+
+        if self.rank_strings:
+            args = [
+                "rank_strings",
+                "--limit",
+                str(self.max_no_of_strings),
+                "--strings",
+                json_dumps(result),
+            ]
+            req_data = {"args": args, "timeout": self.timeout}
+            result = self._docker_run(req_data)
+        elif len(result) > self.max_no_of_strings:
+            result = [s for s in result[: self.max_no_of_strings]]
```

Comment:
I think you forgot to use *max_chars_for_string* attribute. This is extremely important to reduce the possible output